packages/module/src/DataViewTableTree/DataViewTableTree.tsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import {
+  Table,
+  TableProps,
+  Tbody,
+  Td,
+  TdProps,
+  TreeRowWrapper,
+} from '@patternfly/react-table';
+import { useInternalContext } from '../InternalContext';
+import { DataViewTableHeader } from '../DataViewTableHeader';
+import { DataViewTh, DataViewTrTree, isDataViewTdObject } from '../DataViewTable';
+
+export interface DataViewTableTreeProps extends Omit<TableProps, 'onSelect' | 'rows'> {
+  /** Columns definition */
+  columns: DataViewTh[];
+  /** Current page rows */
+  rows: DataViewTrTree[];
+  /** Optinal icon for the leaf rows */
+  leafIcon?: React.ReactNode;
+  /** Optinal icon for the expanded parent rows */
+  expandedIcon?: React.ReactNode;
+  /** Optinal icon for the collapsed parent rows */
+  collapsedIcon?: React.ReactNode;
+  /** Custom OUIA ID */
+  ouiaId?: string;
+}
+
+export const DataViewTableTree: React.FC<DataViewTableTreeProps> = ({
+  columns,
+  rows,
+  leafIcon = null,
+  expandedIcon = null,
+  collapsedIcon = null,
+  ouiaId = 'DataViewTableTree',
+  ...props
+}: DataViewTableTreeProps) => {
+  const { selection } = useInternalContext();
+  const { onSelect, isSelected, isSelectDisabled } = selection ?? {};
+  const [ expandedNodeIds, setExpandedNodeIds ] = React.useState<string[]>([]);
+  const [ expandedDetailsNodeNames, setExpandedDetailsNodeIds ] = React.useState<string[]>([]);
+
+  const getDescendants = (node: DataViewTrTree): DataViewTrTree[] => {
+    if (!node.children || !node.children.length) {
+      return [ node ];
+    } else {
+      let children: DataViewTrTree[] = [];
+      node.children.forEach((child) => {
+        children = [ ...children, ...getDescendants(child) ];
+      });
+      return children;
+    }
+  };
+
+  const areAllDescendantsSelected = (node: DataViewTrTree) => getDescendants(node).every((n) => isSelected?.(n));
+  const areSomeDescendantsSelected = (node: DataViewTrTree) => getDescendants(node).some((n) => isSelected?.(n));
```

Comment:
Memo

Reply:
@Hyperkid123 I'm just not sure what I could use as a dependency for useMemo in these cases

Reply:
Whatever changes the outcomes.